### PR TITLE
Update to Cumulus 28f5ec2

### PR DESCRIPTION
**Pull request by johnwhitton**
Have tested this locally with the latest version.
<details>
  <summary>Sample log showing blocks are being produced</summary>

```
2021-03-28 22:03:24 [Relaychain] ✨ Imported #23 (0xf326…09a7)
2021-03-28 22:03:24 [Parachain] Starting collation. relay_parent=0xf3261d46cfaaeed9c2558970dc89ae139d215744ab647fce7d6ea7c29ab909a7 at=0x7ca5f82aac6c9a61dc0acbf17c53cbd39b30894471b4012d01e652a3b372e06d
2021-03-28 22:03:24 [Parachain] 🙌 Starting consensus session on top of parent 0x7ca5f82aac6c9a61dc0acbf17c53cbd39b30894471b4012d01e652a3b372e06d
2021-03-28 22:03:24 [Parachain] 🎁 Prepared block for proposing at 2 [hash: 0x1f6c029cb3a8ec038b3ae75d7d0ec6de646dd6f884457f3f59c2373cef680708; parent_hash: 0x7ca5…e06d; extrinsics (2): [0x4626…3de5, 0x7b74…cdd5]]
2021-03-28 22:03:24 [Parachain] ✨ Imported #2 (0x1f6c…0708)
2021-03-28 22:03:24 [Parachain] Produced proof-of-validity candidate. pov_hash=0xe1c42b7463d35063037db71017316777f5aeff89869b3ef9b62ce65171e3be06 block_hash=0x1f6c029cb3a8ec038b3ae75d7d0ec6de646dd6f884457f3f59c2373cef680708
2021-03-28 22:03:26 [Relaychain] 💤 Idle (2 peers), best: #23 (0xf326…09a7), finalized #0 (0x186d…c772), ⬇ 1.0kiB/s ⬆ 2.1kiB/s
2021-03-28 22:03:26 [Parachain] 💤 Idle (0 peers), best: #1 (0x7ca5…e06d), finalized #0 (0xdec1…043a), ⬇ 0.3kiB/s ⬆ 0.2kiB/s
2021-03-28 22:03:30 [Relaychain] ✨ Imported #24 (0x2cb1…cc5e)
2021-03-28 22:03:31 [Relaychain] 💤 Idle (2 peers), best: #24 (0x2cb1…cc5e), finalized #0 (0x186d…c772), ⬇ 0.6kiB/s ⬆ 0.3kiB/s
2021-03-28 22:03:31 [Parachain] 💤 Idle (0 peers), best: #1 (0x7ca5…e06d), finalized #0 (0xdec1…043a), ⬇ 0.6kiB/s ⬆ 0.6kiB/s
2021-03-28 22:03:36 [Relaychain] ✨ Imported #25 (0x2666…0733)
```
</details>

### CODE REVIEW

**Review by riusricardo — COMMENTED** [review]
Thanks @johnwhitton LGTM

**Comment by NukeManDan** [comment]
@johnwhitton Awesome! I want to test this locally first to confirm I know it works, then we are GTG.

**Comment by GopherJ** [comment]
FYI, if you run `cargo check` under `runtime` directory it still gives some errors:

image

**Comment by GopherJ** [comment]
also ` cargo check --target wasm32-unknown-unknown --no-default-features` fails while compiling `runtime`

**Comment by GopherJ** [comment]
Hi do you have an idea on this @johnwhitton ?

**Comment by NukeManDan** [comment]
I was able to compile completely, but with warnings (and a few failures from `out of memory` on my machine 😅 - AMD Ryzen 5 3500U, ~12 GB RAM + 1 GB swap):

Polkadot : 94809e63aaf746eb87adf0296614091815c1bf93
this parachain : 042b99f52c254105a2fcaefcff5ea7a2cc4105c1

```
installed targets for active toolchain
--------------------------------------

wasm32-unknown-unknown
x86_64-unknown-linux-gnu

active toolchain
----------------

stable-x86_64-unknown-linux-gnu (default)
rustc 1.51.0 (2fd73fabe 2021-03-23)
```
---

parachain build:
```
   Compiling polkadot-service v0.8.3 (https://github.com/paritytech/polkadot?branch=rococo-v1#2e07b871)
   Compiling cumulus-client-network v0.1.0 (https://github.com/paritytech/cumulus.git?branch=rococo-v1#9d89ed65)
   Compiling polkadot-test-service v0.8.29 (https://github.com/paritytech/polkadot?branch=rococo-v1#2e07b871)
   Compiling cumulus-client-consensus-relay-chain v0.1.0 (https://github.com/paritytech/cumulus.git?branch=rococo-v1#9d89ed65)
   Compiling cumulus-client-collator v0.1.0 (https://github.com/paritytech/cumulus.git?branch=rococo-v1#9d89ed65)
   Compiling cumulus-client-service v0.1.0 (https://github.com/paritytech/cumulus.git?branch=rococo-v1#9d89ed65)
   Compiling polkadot-cli v0.8.29 (https://github.com/paritytech/polkadot?branch=rococo-v1#2e07b871)
   Compiling parachain-collator v2.0.0 (/home/dan/substrate-parachain-template/node)
warning: unused import: `sp_core::Pair`
  --> node/src/service.rs:15:5
   |
15 | use sp_core::Pair;
   |     ^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

warning: unused variable: `task_executor`
   --> node/src/command.rs:268:9
    |
268 |                 let task_executor = config.task_executor.clone();
    |                     ^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_task_executor`
    |
    = note: `#[warn(unused_variables)]` on by default

warning: 2 warnings emitted

    Finished release [optimized] target(s) in 13m 35s
```
---

Ruining Polkadot w/ real-overseer & this parachain (as described in the readme):

```
2021-04-01 16:14:29 [Parachain] 💤 Idle (0 peers), best: #6 (0x28b0…4ecb), finalized #0 (0xa43e…8390), ⬇ 0.3kiB/s ⬆ 0.3kiB/s    
2021-04-01 16:14:30 [Relaychain] ✨ Imported #77 (0x0a77…3364)    
2021-04-01 16:14:30 [Parachain] Starting collation. relay_parent=0x0a773e50aa66d1e052262a32c77f9697618e16c7536ce11339524ea864c43364 at=0x4bc1aa5768cc8c74162693f559bb8a8f1e388aaa5c6030f2dec7c16ed812ec41
2021-04-01 16:14:30 [Parachain] 🙌 Starting consensus session on top of parent 0x4bc1aa5768cc8c74162693f559bb8a8f1e388aaa5c6030f2dec7c16ed812ec41    
2021-04-01 16:14:30 [Parachain] 🎁 Prepared block for proposing at 8 [hash: 0x3fe8ab6144ec4cae0373f577cd0d47f5830cca9698294784dfb339b9ba2ca095; parent_hash: 0x4bc1…ec41; extrinsics (2): [0x8c2a…13b5, 0xbdfe…9247]]    
2021-04-01 16:14:30 [Parachain] Produced proof-of-validity candidate. pov_hash=0xf53c5896ab4d1d8c85f00db31ca3cef984f118df69fe394743a5ee3ef14066bb block_hash=0x3fe8ab6144ec4cae0373f577cd0d47f5830cca9698294784dfb339b9ba2ca095
2021-04-01 16:14:30 [Parachain] ✨ Imported #8 (0x3fe8…a095)    
2021-04-01 16:14:34 [Relaychain] 💤 Idle (2 peers), best: #77 (0x0a77…3364), finalized #26 (0x4222…0ab4), ⬇ 1.0kiB/s ⬆ 2.1kiB/s    
2021-04-01 16:14:34 [Parachain] 💤 Idle (0 peers), best: #7 (0x4bc1…ec41), finalized #0 (0xa43e…8390), ⬇ 0.2kiB/s ⬆ 0.2kiB/s    
2021-04-01 16:14:36 [Relaychain] ✨ Imported #78 (0x73c3…a6fd)    
```

---

So blocks are not finalizing. I did _only_ follow the readme (that you updated, and I will update a bit now. Am I missing something?

**Comment by NukeManDan** [comment]
@johnwhitton just made a commit directly to update the readme 😅  meant to make a PR. You agree with the changes there? 

Still get no finality 🤷🏼

File: README.md [hunk 1]
```diff
@@ -27,24 +27,109 @@ cargo build --release
...
+
+`Developer -> sudo -> paraSudoWrapper -> sudoScheduleParaInitialize(id, genesis)`
+
+Ensure you set the `ParaId to 200` and the `parachain: Bool to Yes`.
+
+The files you will need are in the `./resources` folder, you just created.
+
+#### Restart the Parachain (Collator) and Wait...
+
+The collator node may need to be restarted to get it functioning as expected. After a [new era](https://wiki.polkadot.network/docs/en/glossary#era) starts on the relay chain, your parachain will come online. Once this happens, you should see the
```
Comment (NukeManDan) [inline README.md:111]: @riusricardo am I right in saying you need to wait for an era? or is in an epoch? https://wiki.polkadot.network/docs/en/glossary#epoch

Comment (NukeManDan) [reply README.md:111]: The relay reports a large lag in finality, I think this just got changed in rococo-v1 for polkadot? 
tested with polkadot @  94809e63aaf746eb87adf0296614091815c1bf93

remove training wheels f0d5c3d578689109aecb2660a98432e7cd10f53d
should fix this?

**Comment by NukeManDan** [comment]
CI has failed, but I can confirm it built on my machine (described above)

**Comment by GopherJ** [comment]
> CI has failed, but I can confirm it built on my machine (described above)

Have you tested to run `cargo check` under `runtime` directory? it fails on my side.

```
cargo check
```
and
```
cargo check --target wasm32-unknown-unknown --no-default-features
```
both fail

**Pull request merged**